src/dawn/engine_dawn.cc

```diff
@@ -1166,8 +1212,42 @@ Result EngineDawn::DoDrawArrays(const DrawArraysCommand* command) {
   return map.result;
 }  // namespace dawn
 
-Result EngineDawn::DoCompute(const ComputeCommand*) {
-  return Result("Dawn:DoCompute not implemented");
+Result EngineDawn::DoCompute(const ComputeCommand* command) {
+  Result result;
+
+  ComputePipelineInfo* compute_pipeline = GetComputePipeline(command);
+  if (!compute_pipeline)
+    return Result("DoComput: invoked on invalid or missing compute pipeline");
+
+  ::dawn::ComputePipelineDescriptor computePipelineDescriptor;
+  computePipelineDescriptor.layout = MakeBasicPipelineLayout(
+      device_->Get(), compute_pipeline->bind_group_layouts);
+
+  ::dawn::PipelineStageDescriptor pipelineStageDescriptor;
+  pipelineStageDescriptor.module = compute_pipeline->compute_shader;
+  pipelineStageDescriptor.entryPoint = "main";
```

Comment:
Vulkan allows an empty string here, which then defaults to "main".
I think this is ok. No request to change.

Reply:
Dawn isn't as forgiving. If I do not specify the entry point (remove this line) it errors out:
Device error: Entry point must be "main"

Reply:
I asked, and it turns out this is a temporary limitation in Dawn.  So I filed an issue to enhance this in the future. https://github.com/google/amber/issues/607